File: Difficulty: Easy/Array Duplicates/array-duplicates.js
//{ Driver Code Starts
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split('\n').map(string => { return string.trim(); });
    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine()); // Number of test cases
    for (let i = 0; i < t; i++) {
        let arr = readLine().split(' ').map(x => parseInt(x)); // Input array
        let obj = new Solution();
        const result = obj.findDuplicates(arr);
        result.sort((a, b) => a - b); // Ensure the result is in ascending order
        // Print the result in the required format
        if (result.length > 0) {
            console.log(result.join(' ')); // Print duplicates in ascending order
        } else {
            console.log("[]"); // Print empty list if no duplicates are found
        }
        console.log("~"); // Print empty list if no duplicates are found
    }
}

// } Driver Code Ends


// User function Template for javascript

/**
 * @param {number[]} arr
 * @returns {number[]}
 */

class Solution {
    findDuplicates(arr) {
        // code here
        
        let myArray = []
        let myObject = {}

        for (let element of arr) {
                (!myObject[element]) ? myObject[element] = 1 : myObject[element]++;
        }

        for (let key in myObject) {
                if (myObject[key]> 1    ){
                        myArray.push(key)
                }
        }


        return myArray
    }
}

File: Difficulty: Easy/Array Duplicates/array-duplicates.py

class Solution:
    def findDuplicates(self, arr):
        # code here
        hash_map = {}
        repeatations = []
        for i in arr:
            if i not in hash_map:
                hash_map[i] = 1
            else:
                if hash_map[i] ==1 :
                    repeatations.append(i)
                hash_map[i] += 1
        return repeatations



#{ 
 # Driver Code Starts
# Initial Template for Python 3

t = int(input())  # number of test cases
for _ in range(t):
    arr = list(map(int, input().split()))  # input array
    s = Solution().findDuplicates(arr)  # find the duplicates
    # Sort the result in ascending order
    s.sort()
    # Output formatting
    if s:
        print(" ".join(map(str, s)))  # Print duplicates in ascending order
    else:
        print("[]")  # Print empty list if no duplicates are found
    print("~")

# } Driver Code Ends